src/components/StateFooter/index.tsx

```diff
@@ -1,25 +1,30 @@
 import React, { useContext } from "react";
 import { Box, Button } from "@material-ui/core";
 import { CommentsContext } from "../../context/comments";
-import { CommentsActionType } from "../../common/enums";
+import CommentDialog from "../CommentDialog";
+import CommentsActionType from "../../common/enums/comments-action-type";
 
 const StateFooter = () => {
   const [state, dispatch] = useContext(CommentsContext);
+  const [open, setOpen] = React.useState(false);
 
   console.log(state);
 
-  const handleViewState = (event: any) => {};
+  const handleViewState = () => setOpen(true);
 
   const handleResetState = (event: any) => {
     dispatch({
       type: CommentsActionType.RESET_STATE,
     });
   };
 
+  const handleClose = () => setOpen(false);
+
   return (
     <Box display="flex" justifyContent="space-between" p={2}>
       <Button onClick={handleViewState}>View Current State</Button>
       <Button onClick={handleResetState}>Reset State</Button>
+      <CommentDialog isOpen={open} handleClose={handleClose} />
     </Box>
```

Comment:
The third child of the Box element is causing the `Reset State` button to move to the middle in the UI. This is because the flexbox is ensuring that there is space between each child in the horizontal axis. I suggest you move it outside the Box element.
```suggestion
    <React.Fragment>
      <Box display="flex" justifyContent="space-between" p={2}>
        <Button onClick={handleViewState}>View Current State</Button>
        <Button onClick={handleResetState}>Reset State</Button>
      </Box>
      <CommentDialog isOpen={open} handleClose={handleClose} />
    </React.Fragment>
```